### PR TITLE
Jgfouca/branch for acme split 2018 01 30

### PR DESCRIPTION
**Pull request by jgfouca**
Shared change list:
* bless_test_results fix using REALUSER
* more robust stderr presentation for run_cmd_no_fail

Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes [CIME Github issue #]

User interface changes?: 

Update gh-pages html (Y/N)?:

Code review: @jedwards4b

### CODE REVIEW

**Comment by jgfouca** [comment]
@billsacks , I used --squash for this split. Hope that resolved the history issues.

File: scripts/lib/CIME/bless_test_results.py [hunk 1]
```diff
@@ -43,33 +43,35 @@ def bless_namelists(test_name, test_dir, report_only, force, baseline_name, base
 def bless_history(test_name, test_dir, baseline_name, baseline_root, report_only, force):
 ###############################################################################
     with Case(test_dir) as case:
-        if baseline_name is None:
-            baseline_name = case.get_value("BASELINE_NAME_CMP")
-            if not baseline_name:
-                baseline_name = CIME.utils.get_current_branch(repo=CIME.utils.get_cime_root())
+        real_user = case.get_value("REALUSER")
```
Comment (jgfouca) [inline scripts/lib/CIME/bless_test_results.py]: Bug fix for dynamic USER evaluation breaking bless_test_results.

File: scripts/lib/CIME/case.py [hunk 2]
```diff
@@ -908,6 +908,8 @@ def configure(self, compset_name, grid_name, machine_name=None,
             else:
                 self._check_testlists(compset_alias, grid_name, files)
 
+        self.set_value("REALUSER", os.environ["USER"])
```
Comment (jedwards4b) [inline scripts/lib/CIME/case.py]: this is kinda ugly - who is real user?  The current user, or the one that setup the case?

File: scripts/lib/CIME/utils.py [hunk 3]
```diff
@@ -374,6 +410,17 @@ def run_cmd_no_fail(cmd, input_str=None, from_dir=None, verbose=None,
         # If command produced no errput, put output in the exception since we
         # have nothing else to go on.
         errput = output if not errput else errput
+        if errput is None:
```
Comment (jgfouca) [inline scripts/lib/CIME/utils.py]: Fix some cases where stderr reporting was causing problems.

File: src/drivers/mct/cime_config/config_component.xml [hunk 4]
```diff
@@ -509,7 +517,7 @@
   <entry id="BATCH_SYSTEM">
      <type>char</type>
      <default_value>none</default_value>
-     <valid_values>lc_slurm,moab,pbs,lsf,slurm,cobalt,cobalt_theta,none</valid_values>
+     <valid_values>nersc_slurm,lc_slurm,moab,pbs,lsf,slurm,cobalt,cobalt_theta,none</valid_values>
```
Comment (jedwards4b) [inline src/drivers/mct/cime_config/config_component.xml]: Why so many exceptions?  All on the acme side...

File: scripts/lib/CIME/case.py (hunk 2 above)
Comment (jgfouca) [reply scripts/lib/CIME/case.py]: Yeah, it isn't great. REALUSER is the one who created the case. This fixes the bless_test_results issue we were discussing yesterday without needing additional flags to that tool.

File: src/drivers/mct/cime_config/config_component.xml (hunk 4 above)
Comment (jgfouca) [reply src/drivers/mct/cime_config/config_component.xml]: @jedwards4b , that's a good question. I was not involved in the creation of any of these. @ndkeen , do you know why a nersc_slurm batch system was created? It appears to be an exact copy of slurm.

Comment (jgfouca) [reply src/drivers/mct/cime_config/config_component.xml]: Correction, it looks like nersc_slurm uses -q to select queue and slurm uses -p.

Comment (jedwards4b) [reply src/drivers/mct/cime_config/config_component.xml]: We do that on the cesm side and still maintain a single slurm definition.

Comment (jgfouca) [reply src/drivers/mct/cime_config/config_component.xml]: You're right, we need to clean our stuff up.

**Comment by jgfouca** [comment]
@billsacks , I put the couple inline comments I made into the PR description. There were very few changes beyond ACME config changes.

As far as the file history, I'm at a complete loss on how to address that other than entirely suspending ACME/ESMCI integration.

**Comment by billsacks** [comment]
@jgfouca ah right, now I remember: The plan for fixing the problem didn't have to do with how you created this branch, but has to do with how this branch is merged to master. As long as you merge this to master using `--squash` (on the command-line, or presumably you'd get the same thing with the "squash and merge" option on the "merge pull request" button), then history won't get messed up in this way. The downside is that this will look like a single commit with you as the author.

Will that plan work for you?

I'll go ahead and approve these changes, under the assumption that that's how you'll bring these changes to master.

**Comment by jgfouca** [comment]
@billsacks , yes

**Comment by jgfouca** [comment]
@billsacks , OK, I did a squash-and-merge. See what you think.

**Comment by billsacks** [comment]
I spot-checked `scripts/lib/CIME/hist_utils.py` and at least that one looks good. So I think this squash-and-merge worked. Thank you!

**Pull request merged**